### PR TITLE
Add a method to get maximum heap size for zOS compressedrefs mode

### PR DESCRIPTION
**Pull request by JasonFengJ9**
The heap size is determined via `getUserExtendedPrivateAreaMemoryType()`;
Moved `omriarv64.h` from `port/zos390` to `include_core/unix/zos`;
Added two constants `MAXIMUM_HEAP_SIZE_RECOMMENDED_FOR_COMPRESSEDREFS` & `MAXIMUM_HEAP_SIZE_RECOMMENDED_FOR_3BIT_SHIFT_COMPRESSEDREFS` (moved from `OpenJ9` project)

Related https://github.com/eclipse-openj9/openj9/pull/12741

Signed-off-by: Jason Feng <fengj@ca.ibm.com>

### CODE REVIEW

**Comment by JasonFengJ9** [comment]
@gacholio could you please review?

fyi @dmitripivkine @babsingh @0xdaryl

File: include_core/omrutil.h [hunk 1]
```diff
@@ -353,6 +353,14 @@ void OMRZeroMemory(void *ptr, uintptr_t length);
 */
 void j9memset(void *dest, intptr_t value, uintptr_t size);
 
+#if defined(J9ZOS39064)
+/**
+* @brief get maximum heap size allowed for zOS compressedrefs mode
+* @return the heap size allowed
+*/
+U_64 zosGetMaxHeapSizeForCR();
```
Comment (gacholio) [inline include_core/omrutil.h:361]: Args should explicitly be void.

File: util/omrutil/j9memclr.cpp [hunk 2]
```diff
@@ -29,6 +29,14 @@
 #endif /* defined(J9ZOS390) || (defined(LINUX) && defined(S390)) */
 #include <string.h>
 
+#if defined(J9ZOS39064)
+#include "omrgcconsts.h"
+#include "omriarv64.h"
+#pragma linkage (GETTTT,OS)
+#pragma map (getUserExtendedPrivateAreaMemoryType,"GETTTT")
+UDATA getUserExtendedPrivateAreaMemoryType();
```
Comment (gacholio) [inline util/omrutil/j9memclr.cpp:37]: Explciit void.

File: util/omrutil/j9memclr.cpp [hunk 3]
```diff
@@ -163,3 +171,29 @@ j9memset(void *dest, intptr_t value, uintptr_t size)
 {
 	memset(dest, (int) value, (size_t) size);
 }
+
+#if defined(J9ZOS39064)
+U_64
+zosGetMaxHeapSizeForCR()
```
Comment (gacholio) [inline util/omrutil/j9memclr.cpp:177]: Explicit void.

File: include_core/omrutil.h (hunk 1 above)
Comment (JasonFengJ9) [reply include_core/omrutil.h]: Fixed.

File: util/omrutil/j9memclr.cpp (hunk 3 above)
Comment (JasonFengJ9) [reply util/omrutil/j9memclr.cpp]: Fixed.

File: util/omrutil/j9memclr.cpp (hunk 2 above)
Comment (JasonFengJ9) [reply util/omrutil/j9memclr.cpp]: Fixed.

**Review by gacholio — APPROVED** [review]

**Comment by 0xdaryl** [comment]
Jenkins build all

**Comment by gacholio** [comment]
OSX failure appears infra-related. Window failure is impenetrable.

**Comment by JasonFengJ9** [comment]
Windows failure is infra related as well.
```
13:01:08  ERROR: Cannot delete workspace :Unable to delete 'c:\jenkins\workspace\PullRequest-win_x86-64\Build'. Tried 3 times (of a maximum of 3) waiting 0.1 sec between attempts.
```

**Comment by gacholio** [comment]
That really should not cause a successful build to fail.

**Comment by JasonFengJ9** [comment]
Related https://github.com/eclipse-openj9/openj9/issues/10834

fyi @jdekonin

**Review by 0xdaryl — APPROVED** [review]

**Pull request merged**